### ISSUE TITLE
Give error if earlier arg has default but later arg not

### ISSUE THREAD

**Issue by kkmuffme** [squizlabs/PHP_CodeSniffer#2719]
If a function has multiple args and there is any arg with a default value followed by an arg without default value, it should give an error.
Because the supplied default value is useless, since PHP will throw a fatal if the function is called with the latter arg, so you need to supply the previous arg anyway.

e.g.

```php
function test( $a = 'foo', $b ) {
```

To call this you always need to supply both args to avoid a fatal: `test( 'hello', 'world' )` so the default value of $a is useless

**Comment by gsherwood** [comment]
This is what the `PEAR.Functions.ValidDefaultValue` sniff does, although you wont have this check if you are using PSR2 or PSR12 because this is not a specific requirement of the standard.

If you have a custom ruleset.xml file you are using to define your own standard, include that PEAR sniff and the sample code you included will generate the error `Arguments with default values must be at the end of the argument list`

**Comment by kkmuffme** [comment]
Thanks a lot
